src/app/api/quality_matrix/models.py

```diff
@@ -1,18 +1,38 @@
+from dataclasses import dataclass
 from enum import Enum
-from typing import Union
+from typing import Optional
 
 from pydantic import BaseModel, Field
 from sqlalchemy import JSON, Column, Integer, Text
 from sqlalchemy.orm import declarative_base
 
 
-class ColumnOutputModel(BaseModel):
+@dataclass(frozen=True)
+class QualityOutput:
+    metadatum: str
+    level: Optional[
+        int
+    ]  # Needs to be optional since timeline database does not have levels - yet
+    columns: dict[str, float]
+
+
+class QualityOutputModel(BaseModel):
     metadatum: str = Field(default="", description="Name of the evaluated metadatum.")
+    level: int = Field(default=1, description="Hierarchy level of this metadatum")
     columns: dict[str, float] = Field(
         description="The ratio of quality for the required columns."
     )
 
 
+class QualityOutputResponse(BaseModel):
```

Comment:
Proposal:
 - rename this to `QualityOutput` (or `QualityResponse` depending on whether the respective input is named `QualityRequest` or `QualityInput`)
 - rename `QualityOutputModel` to `QualityRow`
 - remove `QualityOutput` dataclass and use `QualityRow` instead.

Reply:
I added this to this issue: https://github.com/openeduhub/metaqs-main/issues/72
I like the suggestion, but those would be bigger changes and this PR is already rather big.